data/ai/lua/ca_recruit_rushers.lua

```diff
@@ -944,6 +944,11 @@ function ca_recruit_rushers:evaluation(cfg, data, filter_own)
         end
     end
 
+    recruit_data.recuit_types = wesnoth.sides[wesnoth.current.side].recruit
+    for _,unit_type in ipairs(leader.extra_recruit) do
+        table.insert(recruit_data.recuit_types, unit_type)
```

Comment:
This might cause duplicate entries if both wesnoth.sides[wesnoth.current.side].recruit and leader.extra_recruit contain the same unit.
Apart from slowing things down, depending on the level of randomness, it could also increase the chance that unit is selected since it gets two chances to have the highest score, not one.